src/token_contract/src/types/token_utils/transferrable.nr

```diff
@@ -0,0 +1,179 @@
+pub global TRANSFERRABLE_CODE: Quoted = quote {
+    
+    global INITIAL_TRANSFER_CALL_MAX_NOTES: u32 = 2;
+    global RECURSIVE_TRANSFER_CALL_MAX_NOTES: u32 = 8;
+    
+    /** ==========================================================
+    * ========================= PRIVATE =========================
+    * ======================================================== */
+
+    #[private]
+    fn transfer_private_to_public(
+        from: AztecAddress,
+        to: AztecAddress,
+        amount: U128,
+        nonce: Field,
+    ) {
+        TokenLib::_validate_from_private(from, nonce, &mut context);
+
+        TokenLib::_decrease_private_balance(
+            &mut context,
+            storage.private_balances,
+            from,
+            amount,
+            INITIAL_TRANSFER_CALL_MAX_NOTES,
+        )
+            .emit(encode_and_encrypt_note(&mut context, from, from));
+
+        // let self = Token::at(context.this_address());
+        // self.increase_public_balance(to, amount).enqueue(&mut context);
+        context.call_public_function(
+            context.this_address(),
+            FunctionSelector::from_signature("increase_public_balance((Field),(Field,Field))"),
+            [to.to_field(), amount.to_field()],
+        );
```

Comment:
i think `call_public_function` should belong to TokenLib and not the token code implementation, we should keep on abstracting TokenLib to have these files as simple as possible

token implementation should:
- [ ] allow different implementations of log encrypting and authwit logic (i.e. override `validate_from` method)
- [ ] emit all logs (since it needs to be able to override the `encode_and_encrypt_note`)
- [ ] depend on TokenLib to do practically anything, it just shapes the `storage.public_balances` into the right call arguments
- [ ] self.call (enqueues) should be handled by TokenLib, never here in token contract
- [ ] TokenLib should have all methods available, missing `(private_)increase_public_balance`